### PR TITLE
docs (bodiless-documentation): Edit search component documentation

### PR DESCRIPTION
**Pull request by malachye**
## Changes
- Edit and reformat Search documentation to include information for Content Editors, Site Builders and Architectural Details

## Related Issues:
#693 
https://jira.jnj.com/browse/AESQ-3637

### CODE REVIEW

File: packages/bodiless-search/README.md [hunk 1]
```diff
@@ -1,73 +1,159 @@
...
+### 3. Support Search Indexing
+
+In this step, you will update `package.json` to build the search index. In
+addition, you will add search index command that you can run on demand. A search
+index command will be added to build step as well so that it is updated
+automatically on every build.
+
+_Make the following changes (Steps 3a, 3b, 3c) in the `scripts` section of `package.json`._
+
+#### 3a. Create a search index task
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:94]:
```suggestion
#### 3a. Create a `search-index` task
```

Comment (hvanyo) [reply packages/bodiless-search/README.md:94]: For consistency https://edit.kjsjrzigky-jvaff4bu65vgm.us-2.platformsh.site/___docs/#/Components/Search to match the others

File: packages/bodiless-search/README.md [hunk 2]
```diff
@@ -1,73 +1,159 @@
...
+This will create the search index under path specified by
+`BODILESS_SEARCH_INDEX_PATH` from configuration.
+
+### 4. Create Search HOCs
 
-### Create Styled Search Components
+Create the search HOC functions, which will add and define the styles for search
+comopnents.
+
+Under site project root, add a new folder as src/component/Search for site level
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:130]:
```suggestion
Under site project root, add a new folder as `src/component/Search` for site level
```

File: packages/bodiless-search/README.md [hunk 3]
```diff
@@ -129,38 +222,157 @@ const HeaderClean: FC<Props> = ({ components }) => {
     <Wrapper>
       <Container>
         <SiteLogoReturn />
-        <SimpleSearchBox />
+        <SimpleSearchBox placeholder ="Search" />
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:225]:
```suggestion
        <SimpleSearchBox placeholder="Search" />
```

File: packages/bodiless-search/README.md [hunk 4]
```diff
@@ -129,38 +222,157 @@ const HeaderClean: FC<Props> = ({ components }) => {
...
+      <PrimaryHeader/>
       <InlineSearchBox />
-      <SimpleSearchResult />
+      <SimpleSearchResult resultCountMessage={myCountMessage} resultEmptyMessage={myEmptyMessage} />
     </SearchLayout>
   </Page>
 );
 ```
 
+?> `resultCountmessage` & `resultEmptyMessage` defaults to const `ResultCountMessage = 'Showing %count% result(s).';
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:265]: This is going outside the width of the page: https://edit.kjsjrzigky-jvaff4bu65vgm.us-2.platformsh.site/___docs/#/Components/Search?id=_6-add-search-results-page-using-the-simplesearchresult-component Can you play with it and see if you can it format within.  force lines.

File: packages/bodiless-search/README.md [hunk 5]
```diff
@@ -129,38 +222,157 @@ const HeaderClean: FC<Props> = ({ components }) => {
...
+const myCountMessage = "Demostración %count% Resultados";
 
 const SearchPage = (props: any) => (
   <Page {...props}>
     <SearchLayout>
-      <h1 className="text-3xl font-bold">Search Result</h1>
+      <PrimaryHeader/>
       <InlineSearchBox />
-      <SimpleSearchResult />
+      <SimpleSearchResult resultCountMessage={myCountMessage} resultEmptyMessage={myEmptyMessage} />
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:259]:
```suggestion
        <SimpleSearchResult
          resultCountMessage={myCountMessage}
          resultEmptyMessage={myEmptyMessage}
        />
```

Comment (hvanyo) [reply packages/bodiless-search/README.md:259]: because its overflowin the code box.

File: packages/bodiless-search/README.md [hunk 6]
```diff
@@ -129,38 +222,157 @@ const HeaderClean: FC<Props> = ({ components }) => {
...
+search result component and strings utilize the count token (%count%) which
+substitutes a value of search results found.
+
+
+> Note: for complete search component implementation example, please check 
+[Test-Site/Search Component](https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/examples/test-site/src/components/Search/index.tsx).
+
+### Site Build Troubleshooting Guide.
 
+The site search index is cached in browser cache so if you make changes, they
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:278]:
```suggestion
The site search index is cached in your browser cache so if you make changes, they
```

File: packages/bodiless-search/README.md [hunk 7]
```diff
@@ -129,38 +222,157 @@ const HeaderClean: FC<Props> = ({ components }) => {
...
+1. Click Application tab
+2. Click Local Storage and your site
+3. Then clear search:index file.
 
-> Note: for complete search component implementation example, please check [Test-Site/Search Component](https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/examples/test-site/src/components/Search/index.tsx).
+### Defining What is Indexed
 
+Indexing on your site is defined by the `BODILESS_SEARCH_INDEX_SELECTOR` env
+variable. It is recommended that you set it to the body of your content or body
+of your articles. It accepts classes or ids or other items and is comma
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:290]:
```suggestion
of your articles. It accepts classes or ids or other selectors and is comma
```

File: packages/bodiless-search/README.md [hunk 8]
```diff
@@ -129,38 +222,157 @@ const HeaderClean: FC<Props> = ({ components }) => {
...
+separated. If you want to index all items in the body of your page use "body *"
+and all content within body will be indexed.
 
+For example, if you want to index article content, you can use ".article-content *" 
+and target article content specifically.
 
+In addition, you can exclude items from search via the env var `BODILESS_SEARCH_INDEX_EXCLUDE_SELECTOR`.
+By default we suggest adding script,noscript, style so this isn't indexed.
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:298]:
```suggestion
By default we suggest adding `script`, `noscript`, `style` so these aren't indexed. 
```

File: packages/bodiless-search/README.md [hunk 9]
```diff
@@ -129,38 +222,157 @@ const HeaderClean: FC<Props> = ({ components }) => {
...
 
+For example, if you want to index article content, you can use ".article-content *" 
+and target article content specifically.
 
+In addition, you can exclude items from search via the env var `BODILESS_SEARCH_INDEX_EXCLUDE_SELECTOR`.
+By default we suggest adding script,noscript, style so this isn't indexed. 
+
+#### Adding a `no-search` Class
+
+Using a `no-search` class anything you do not want indexed can be wrapped in this class.
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:302]:
```suggestion
Using a `no-search` HOC around anything you do not want indexed is very useful.```

File: packages/bodiless-search/assets/.gitignore [hunk 10]
```diff
@@ -0,0 +1 @@
+
```
Comment (dewen) [inline packages/bodiless-search/assets/.gitignore:1]: Do we need this gitignore?

Comment (malachye) [reply packages/bodiless-search/assets/.gitignore]: deleted

File: packages/bodiless-search/README.md [hunk 11]
```diff
@@ -1,73 +1,159 @@
...
+The BodilessJS Search component provides a search box which allows users to
+search for site content. It also provides search results components to display
+results on a designated search page.
 
-## Installation
+![](./assets/search.jpg)
 
-First, install the Bodiless search package:
+?> The BodilessJS Search Component requires search index to be built prior to
+searching.  The Search Component is intended for use in a static environment
```
Comment (dewen) [inline packages/bodiless-search/README.md:10]: Double spaces before "The Search Component ..."

File: packages/bodiless-search/README.md [hunk 12]
```diff
@@ -1,73 +1,159 @@
...
 
-## Specify the url of search result page. The search component will redirect user to this page if current page is different.
+## This is the URL path of the search results landing page. 
+## When you create this page, you must use this same page path when you ## perform step 6 - 
+## "Add search results pages..."
 BODILESS_SEARCH_PAGE='search'
 
-## Set in-browser search index expiration time period in seconds. By default, the index expires in one hour (86400 seconds) from time of load.
+## Set in-browser search index expiration time period in seconds. 
+## By default, the index expires in one hour (86400 seconds) from time  of load.
```
Comment (dewen) [inline packages/bodiless-search/README.md:66]: extra space before "of load"

**Review by dewen — APPROVED** [review]

**Comment by hvanyo** [comment]
@malachye  there are still items running to the outside the righthand side of document -> https://edit.kjsjrzigky-jvaff4bu65vgm.us-2.platformsh.site/___docs/#/Components/Search  some in the codeblocks and some in the notes.

File: packages/bodiless-search/README.md [hunk 13]
```diff
@@ -129,38 +222,161 @@ const HeaderClean: FC<Props> = ({ components }) => {
...
+```
+<NoSearch>
+   <FlowContainerDefault nodeKey="notfound" />
+</NoSearch>
+```
+
+---
+
+## Architectural Details
+
```
Comment (hvanyo) [inline packages/bodiless-search/README.md:354]: Preface this with a link to Bodiless search src code is https://github.com/johnsonandjohnson/Bodiless-JS/tree/master/packages/bodiless-search/src

**Review by dewen — DISMISSED** [review]

**Review by hvanyo — APPROVED** [review]

**Review by dewen — APPROVED** [review]

**Pull request merged**